pysb/testing/modeltests.py

```diff
@@ -0,0 +1,389 @@
+from __future__ import print_function
+import pysb
+import abc
+from pysb.core import SelfExporter
+from pysb.pattern import SpeciesPatternMatcher, ReactionPatternMatcher, \
+    RulePatternMatcher
+
+
+class ModelAssertionFailure(Exception):
+    def __init__(self, assertion, model, message=None):
+        self.assertion = assertion
+        self.model = model
+        self.message = message
+
+    def __repr__(self):
+        base_msg = '%s on model %s failed' % (self.assertion,
+                                              self.model.name)
+        if self.message:
+            return base_msg + ': ' + str(self.message)
+        else:
+            return base_msg
+
+    def __str__(self):
+        return repr(self)
+
+
+class TestSuite(object):
```

Comment:
How would one use this in an actual test suite? `assert ts.check(...)` ?

Reply:
Yes. In principle, the same assertion could be checked against two test suite instances, e.g. add a Rule, regenerate the network, create a new test suite, and compare `ts1.check(my_assertion)` to `tst2.check(my_assertion)`.

One alternative might be `ts.assert(...)` instead of `assert ts.check(...)`. Do you think that'd be a syntactic improvement?